crypto/riscv32cpuid.pl

```diff
@@ -94,8 +94,24 @@
 .globl riscv_vlen_asm
 .type riscv_vlen_asm,\@function
 riscv_vlen_asm:
-    csrr $ret, vlenb
-    slli $ret, $ret, 3
+    # Check for vector extension support.
+    csrrs   $t0,misa,x0
+    andi    $t0,$t0,(1<<20)
+    beqz    $t0,no_vector_extension
+
+    # vlenb is defined by RISC-V "V" Vector Extension
+    # version 1.0.
+    # See: https://github.com/riscv/riscv-v-spec/blob/master/v-spec.adoc
+    .equ    vlenb,0xC22
```

Comment:
I'm still not sure we need to define this in this way.  I'm looking through the gcc manual and our config files and note that we set the compiler arch to riscv[64|32] for all RISC systems.  I'm wondering if we should be setting it to rv[32|64]gv instead to enable the vector extensions, so that gcc properly understands what the vlenb register is.

Reply:
With gcc 14 and `-march=rv64gcv`, checking the disassembly of libcrypto, functions like `ossl_i2c_ASN1_INTEGER`, `ossl_c2i_ASN1_INTEGER`, `ASN1_OBJECT_free` and many others are vectorized.

Reply:
Well, the tag makes sense, no, as the code is targeted for rv64gcv with this approach.  I'm not sure what (adverse) affects you would expect with the linker/disassembler here.

If the goal isn't to generally vectorize the entire build, then it seems we should find a way to add cflags to individual files to specify the arch.  But at the end of the day, it seems to me like the right thing to do is, if we are using features of a given arch, we should tell the compiler that is the arch we are building for.